### ISSUE TITLE
segments.i3wm.mode 

### ISSUE THREAD

**Issue by ferdinandyb** [powerline/powerline#2054]
Adding the following to the config json  (I'm running it with lemonbar, but I guess it should not matter):

```
      {
        "function": "powerline.segments.i3wm.mode",
        "args": {
            "names": {
                "default": "default"
              }
          }
      },
```
results in the following error

```
2019-11-25 21:47:39,439:ERROR:wm:mode:Failed to set highlight group: u'Highlighting groups not found in colorscheme: mode'
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/powerline/segment.py", line 151, in set_segment_highlighting
    segment.get('gradient_level')
  File "/usr/local/lib/python2.7/dist-packages/powerline/colorscheme.py", line 105, in get_highlighting
    raise KeyError('Highlighting groups not found in colorscheme: ' + ', '.join(groups))
KeyError: u'Highlighting groups not found in colorscheme: mode'
```

As far as I can see the issue is that it is missing from `config_files/colorschemes/default.json`. For test purposes, I copied the mode line from the `vim` config, so it now displays "default", but changing modes doesn't have any effect, it display "default" in all modes, which suggests that mode information from `i3` never reaches powerline.

**Comment by PH111P** [comment]
For modes working in i3, you need to pass `--i3` as an argument to the lemonbar binding. 

No `mode` color being defined in the default config is indeed something that should be fixed though.

**Comment by priestoferis** [comment]
This is how I run it: 
```
python /usr/local/lib/python2.7/dist-packages/powerline/bindings/lemonbar/powerline-lemonbar.py -i3 - -f "Source Code Pro for Powerline" -f "Font Awesome"
```

Just to make sure we are talking about the same thing (I am new to `i3`), I have this in my config of `i3`

```
set $mode_launcher test

mode "$mode_launcher" {
    bindsym f exec firefox
    bindsym t exec thunderbird

    bindsym Escape mode "default"
    bindsym Return mode "default"
}

bindsym $mod+v mode "$mode_launcher"
```
and even tried to add this `test` mode to my powerline config to no avail.

```
      {
        "function": "powerline.segments.i3wm.mode",
        "args": {
            "names": {
                "default": "default",
                "test": "i"
              }
          }
      },
```

**Comment by priestoferis** [comment]
My bad, I was using `-i3` instead of `--i3`.

**Comment by priestoferis** [comment]
The following are also missing: 
```
u'Highlighting groups not found in colorscheme: w_focused, w_visible, workspace'
```
so I guess the entire `i3wm` segment group is missing the highlighting by default.

**Comment by gbraad** [comment]
> `u'Highlighting groups not found in colorscheme: w_focused, w_visible, workspace'`

Experiencing the same now

**Comment by gbraad** [comment]
I noticed that

https://github.com/powerline/powerline/tree/develop/powerline/config_files/colorschemes

and 

https://github.com/PH111P/powerline/tree/11094650ee6401dfa8912206ee43e2407f535bd6/powerline/config_files/colorschemes

are different. The upstream does not have a `wm` colorscheme file.

**Comment by gbraad** [comment]
Adding at least:

```
{
    "name": "Default",
    "groups": {
        "workspace":                 { "fg": "white",            "bg": "gray4",           "click": {
            "left": "i3-msg workspace {0:.2}",
            "right": "i3-msg move to workspace {0:.2}"
        } },
        "workspace:focused":         { "fg": "white",            "bg": "gray4" },
        "workspace:visible":         { "fg": "white",            "bg": "gray9" },
        "workspace:urgent":          { "bg": "white",            "fg": "red",           "click": {
            "left": "i3-msg workspace {0:.2}",
            "right": "i3-msg move to workspace {0:.2}"
        } }
    }
}
```

will make it work

**Comment by PH111P** [comment]
> Adding at least:
> 
> ```json
> {
>     "name": "Default",
>     "groups": {
>         "workspace":                 { "fg": "white",            "bg": "gray4",           "click": {
>             "left": "i3-msg workspace {0:.2}",
>             "right": "i3-msg move to workspace {0:.2}"
>         } },
>         "workspace:focused":         { "fg": "white",            "bg": "gray4" },
>         "workspace:visible":         { "fg": "white",            "bg": "gray9" },
>         "workspace:urgent":          { "bg": "white",            "fg": "red",           "click": {
>             "left": "i3-msg workspace {0:.2}",
>             "right": "i3-msg move to workspace {0:.2}"
>         } }
>     }
> }
> ```
> 
> will make it work

Note that for this repo, the `click` field are ignored as this is one of the unmerged features of the fork that you mentioned.

**Comment by gbraad** [comment]
@PH111P I noticed this. Anything that I can do to help?

**Comment by gbraad** [comment]
Even when changed to:
```
        "workspace:focused":         { "fg": "white",            "bg": "darkblue" },
```
the focused workspace does not get highlighted. What could be the cause of this?

**Comment by gbraad** [comment]
```json
{                                                                                                                       │ 154
    "name": "Default",                                                                                                  │ 155 # Move focus between screens
    "groups": {                                                                                                         │ 156 bindsym $mod+Ctrl+h focus output left
        "workspace":         { "fg": "white",            "bg": "gray4" },                                               │ 157 bindsym $mod+Ctrl+l focus output right
        "w_focused":         { "fg": "white",            "bg": "darkblue" },                                            │ 158 bindsym $mod+Ctrl+Left focus output left
        "w_visible":         { "fg": "white",            "bg": "gray9" }                                                │ 159 bindsym $mod+Ctrl+Right focus output right
    }                                                                                                                   │ 160
}
```

**Comment by PH111P** [comment]
> Even when changed to:
> 
> ```json
>         "workspace:focused":         { "fg": "white",            "bg": "darkblue" },
> ```
> 
> the focused workspace does not get highlighted. ~What could be the cause of this?~
> 
> The names are different

Ah sorry, I forgot that. Indeed, the names in upstream are not yet “fixed”, see #1520.

**Comment by PH111P** [comment]
Fixed. Also added an active window segment.
